tensorflow_quantum/python/optimizers/utils.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utils for tensorflow quantum optimizers"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from operator import mul
+from functools import reduce
+import tensorflow as tf
+
+
+def function_factory(model, loss, train_x, train_y):
+    """A factory to create a function required by tfq.optimizer.rotosolve.
+
+    This function is originally defined for l-bgfs minimizer for tensorflow

```

Comment:
This seems like you are providing a more general feature for the library that something that is just for rotosolve. If this is copied from l-bfgs can you provide a link to it ? If it is similar enough perhaps we don't need to write it ourselves and can instead depend on the implementation from tf.

Reply:
This function is introduced from internet https://pychao.com/2019/11/02/optimize-tensorflow-keras-models-with-l-bfgs-from-tensorflow-probability/ but not inside the TensorFlow probability package.